modules/renter/hostdb/hostdb.go

```diff
@@ -238,3 +272,41 @@ func (hdb *HostDB) Host(spk types.SiaPublicKey) (modules.HostDBEntry, bool) {
 func (hdb *HostDB) RandomHosts(n int, excludeKeys []types.SiaPublicKey) []modules.HostDBEntry {
 	return hdb.hostTree.SelectRandom(n, excludeKeys)
 }
+
+// IncrementSuccessfulInteractions increments the number of successful interactions with a host for a given key
+func (hdb *HostDB) IncrementSuccessfulInteractions(key types.SiaPublicKey) {
+	hdb.mu.Lock()
+	defer hdb.mu.Unlock()
+
+	host, haveHost := hdb.hostTree.Select(key)
+	if !haveHost {
+		return
+	}
+
+	// Update historic values if necessary
+	updateHostsHistoricInteractions(&host, hdb.cs.Height())
+
+	// Increment the successful interactions
+	host.RecentSuccessfulInteractions++
+
+	hdb.hostTree.Modify(host)
+}
+
+// IncrementFailedInteractions increments the number of failed interactions with a host for a given key
+func (hdb *HostDB) IncrementFailedInteractions(key types.SiaPublicKey) {
+	hdb.mu.Lock()
+	defer hdb.mu.Unlock()
+
+	host, haveHost := hdb.hostTree.Select(key)
+	if !haveHost {
+		return
+	}
```

Comment:
Change to `if !haveHost || !hdb.online {` - this will prevent the host from being docked due to connection issues on the renter side. It's not foolproof b/c the `online` value is only updated every 30 seconds or so, but if we ever update the `online` value to be more responsive, we'll get the benefit automatically.